l10n_es_aeat/tests/test_l10n_es_aeat_mod_base.py

```diff
@@ -0,0 +1,214 @@
+                 '600000', '700000', '430000', '410000'}
+        for code in codes:
+            self.accounts[code] = self.env['account.account'].search([
+                ('company_id', '=', self.company.id),
+                ('code', '=', code),
+            ])
+        return True
+
+    def _print_move_lines(self, lines):
+        _logger.debug(
+            '%8s %9s %9s %14s %s',
+            'ACCOUNT', 'DEBIT', 'CREDIT', 'TAX', 'TAXES')
+        for line in lines:
+            _logger.debug(
+                '%8s %9s %9s %14s %s',
+                line.account_id.code, line.debit, line.credit,
+                line.tax_line_id.description,
+                line.tax_ids.mapped('description'))
+
+    def _print_tax_lines(self, lines):
+        for line in lines:
+            _logger.debug(
+                "=== [%s] ============================= [%s]",
+                line.field_number, line.amount)
+            self._print_move_lines(line.move_line_ids)
+
+    def _invoice_sale_create(self, dt):
+        data = {
+            'company_id': self.company.id,
+            'partner_id': self.customer.id,
+            'date_invoice': dt,
+            'type': 'out_invoice',
+            'account_id': self.customer.property_account_receivable_id.id,
+            'journal_id': self.journal_sale.id,
+            'invoice_line_ids': [],
+        }
+        _logger.debug('Creating sale invoice: date = %s' % dt)
+        if self.debug:
+            _logger.debug('%14s %9s' % ('SALE TAX', 'PRICE'))
+        for desc, values in self.taxes_sale.iteritems():
+            if self.debug:
+                _logger.debug('%14s %9s' % (desc, values[0]))
+            tax = self.env['account.tax'].search([
+                ('company_id', '=', self.company.id),
+                ('description', '=', desc),
+            ])
+            data['invoice_line_ids'].append((0, 0, {
+                'name': 'Test for tax %s' % desc,
+                'account_id': self.accounts['700000'].id,
+                'price_unit': values[0],
+                'quantity': 1,
+                'invoice_line_tax_ids': [(6, 0, [tax.id])],
+            }))
+        inv = self.env['account.invoice'].create(data)
+        inv.action_invoice_open()
+        if self.debug:
+            self._print_move_lines(inv.move_id.line_ids)
+        return inv
+
+    def _invoice_purchase_create(self, dt):
+        data = {
+            'company_id': self.company.id,
+            'partner_id': self.supplier.id,
+            'date_invoice': dt,
+            'type': 'in_invoice',
+            'account_id': self.customer.property_account_payable_id.id,
+            'journal_id': self.journal_purchase.id,
+            'invoice_line_ids': [],
+        }
+        _logger.debug('Creating purchase invoice: date = %s' % dt)
+        if self.debug:
+            _logger.debug('%14s %9s' % ('PURCHASE TAX', 'PRICE'))
+        for desc, values in self.taxes_purchase.iteritems():
```

Comment:
Para quefuncionen los tests del módulo l10n_es_aeat_mod11 se necesita cambiar esta línea por...
for desc, values in self.taxes_purchase.items():

Los tests del módulo l10n_es_aeat_mod11

Reply:
Si pasáis el comando `2to3` esto lo hace automático, tal como dice la guía de migración: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0

Reply:
OK, voy a esperar que termine el travis.

Reply:
Pasado el `2to3` ha cambiado eso que pedía @javierjcf